docs/middleware.rst

```diff
@@ -381,9 +381,9 @@ retrieved using JSON-RPC endpoints that don't rely on server state.
 .. code-block:: python
```

Comment:
Can we update these to be doctests so we catch these types of errors in CI.  Probably good to revert the changes and confirm the doctest failures and then re-apply the fixes to ensure that doctest is doing what it's supposed to.